src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -390,6 +405,180 @@ def tropical_variety(self):
             return TropicalSurface(self)
         return TropicalVariety(self)
 
+    def newton_polytope(self):
+        """
+        Return the Newton polytope of ``self``.
+
+        The Newton polytope is the convex hull of all the points
+        corresponding to the exponents of the monomials of tropical
+        polynomial.
+
+        OUTPUT: :class:`~sage.geometry.polyhedron.constructor.Polyhedron`
+
+        EXAMPLES:
+
+        A Newton polytope for a two-variable tropical polynomial::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = x + y
+            sage: p1.newton_polytope()
+            A 1-dimensional polyhedron in ZZ^2 defined as the convex hull of 2 vertices
```

Comment:
You should also have this output the data of the Newton polytope here and in the example below.